File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-8.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
    if(s.length() == 0){
        return 0;
    }

    int left = 0; 
    int right = 0;
    int maxWindow = 0; 
    Set<Character> seenSet = new HashSet<>(); 

    while(right < s.length()){ 
        char curr = s.charAt(right);
        if(seenSet.contains(curr)){ 
            seenSet.remove(s.charAt(left)); 
            left++; 
        } else { 
            seenSet.add(curr); 
            maxWindow = Math.max(maxWindow, right - left + 1); 
            right++; 
            
        }
    }

    return maxWindow;


    }
}

File: Data Structures & Algorithms/eating-bananas/submission-8.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        res = 0 
        left = 1 
        right = max(piles) 

        while(left != right):
            mid = left + (right - left)//2 
            timeTaken = 0 

            for pile in piles: 
                timeTaken += math.ceil(pile / mid)

            if(timeTaken > h): 
                left = mid + 1 
                continue 
            else:
                right = mid 

        return left




